api/v1/metric_routes.go

```diff
@@ -36,7 +36,10 @@ func handleGetMetrics(rw http.ResponseWriter, r *http.Request) {
 		t = engine.ExecutionScheduler.GetState().GetCurrentTestRunDuration()
 	}
 
+	engine.MetricsLock.Lock()
 	metrics := newMetricsJSONAPI(engine.Metrics, t)
+	engine.MetricsLock.Unlock()
```

Comment:
Checking the code it makes sense but it's strange that here we didn't see a data race before. Do we know the usage of this endpoint? We should check that we are not impacting the engine performances.

Reply:
On top of my head, I have no answer about the usage of this endpoint, but I guess that this endpoint is just a nice feature.

Usually, external outputs are used for the metric collection.